lib/transport/receiver/xhr.js

```diff
@@ -37,12 +37,13 @@ XhrReceiver.prototype._chunkHandler = function(status, text) {
     return;
   }
 
-  for (var idx = -1; ; this.bufferPosition += idx + 1) {
+  for (var idx = -1; ; ) {
     var buf = text.slice(this.bufferPosition);
     idx = buf.indexOf('\n');
     if (idx === -1) {
       break;
     }
+    this.bufferPosition += idx + 1;
```

Comment:
What do you think about having the `this.emit` wrapped in a try/finally, like you did for JSONP?

Reply:
Sure--I'm happy to do it that way if you prefer. Thanks for looking.